pkg/manager/impl/description_entity_manager.go

```diff
@@ -0,0 +1,181 @@
+	repoInterfaces "github.com/flyteorg/flyteadmin/pkg/repositories/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/transformers"
+	runtimeInterfaces "github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/contextutils"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"google.golang.org/grpc/codes"
+)
+
+type DescriptionEntityMetrics struct {
+	Scope promutils.Scope
+}
+
+type DescriptionEntityManager struct {
+	db      repoInterfaces.Repository
+	config  runtimeInterfaces.Configuration
+	metrics DescriptionEntityMetrics
+}
+
+func createDescriptionEntity(ctx context.Context, db repoInterfaces.Repository, descriptionEntity *admin.DescriptionEntity, id core.Identifier) error {
+	descriptionDigest, err := util.GetDescriptionEntityDigest(ctx, descriptionEntity)
+	if err != nil {
+		logger.Errorf(ctx, "failed to compute description entity digest for [%+v] with err: %v", id, err)
+		return err
+	}
+
+	existingDescriptionEntityModel, err := util.GetDescriptionEntityModel(ctx, db, id)
+	if err == nil {
+		if bytes.Equal(existingDescriptionEntityModel.Digest, descriptionDigest) {
+			return errors.NewFlyteAdminErrorf(codes.AlreadyExists,
+				"identical description entity already exists with id %v", id)
+		}
+
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument,
+			"description entity with different structure already exists with id %v", id)
+	}
+
+	descriptionModel, err := transformers.CreateDescriptionEntityModel(descriptionEntity, id, descriptionDigest)
+	if err != nil {
+		logger.Errorf(ctx,
+			"Failed to transform description model [%+v] with err: %v", descriptionEntity, err)
+		return err
+	}
+
+	var descriptionID uint
+	if descriptionID, err = db.DescriptionEntityRepo().Create(ctx, descriptionModel); err != nil {
+		logger.Errorf(ctx, "Failed to create description model with id [%+v] with err %v", id, err)
+		return err
+	}
+
+	if id.ResourceType == core.ResourceType_TASK {
+		err = db.TaskRepo().UpdateDescriptionID(models.Task{
+			TaskKey: models.TaskKey{
+				Project: descriptionModel.Project,
+				Domain:  descriptionModel.Domain,
+				Name:    descriptionModel.Name,
+				Version: descriptionModel.Version,
+			},
+			DescriptionID: descriptionID,
+		})
+		if err != nil {
+			logger.Errorf(ctx, "Failed to update descriptionID in tasks table: %v", err)
+			return err
+		}
+	} else if id.ResourceType == core.ResourceType_WORKFLOW {
+		err = db.WorkflowRepo().UpdateDescriptionID(models.Workflow{
+			WorkflowKey: models.WorkflowKey{
+				Project: descriptionModel.Project,
+				Domain:  descriptionModel.Domain,
+				Name:    descriptionModel.Name,
+				Version: descriptionModel.Version,
+			},
+			DescriptionID: descriptionID,
+		})
+		if err != nil {
+			logger.Errorf(ctx, "Failed to update descriptionID in workflows table: %v", err)
+			return err
+		}
+	}
+
+	return nil
```

Comment:
do we need to offload the long description somewhere in this method body?

Reply:
we offload the long description in the flytekit side now. Let me know if it's better to do it in flyteadmin.

Reply:
So any flyteadmin service client will have to make sure they similarly offload descriptions? This seems brittle, especially if we have to worry about read access being delegated to client code but have admin responsible for serving up the descriptions to any caller